src/main/java/com/rultor/agents/IndexesRequests.java

```diff
@@ -29,18 +29,99 @@
  */
 package com.rultor.agents;
 
+import com.google.common.base.Function;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Ordering;
+import com.jcabi.xml.XML;
 import com.rultor.spi.SuperAgent;
+import com.rultor.spi.Talk;
 import com.rultor.spi.Talks;
 import java.io.IOException;
+import java.util.List;
+import java.util.UUID;
+import org.xembly.Directives;
 
 /**
  * Adds index to all the requests received.
+ *
  * @author Krzysztof Krason (Krzysztof.Krason@gmail.com)
  * @version $Id$
  */
+@SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
 public final class IndexesRequests implements SuperAgent {
     @Override
     public void execute(final Talks talks) throws IOException {
-        throw new UnsupportedOperationException();
+        if (talks == null) {
+            return;
+        }
+        int idx = this.index(talks);
+        for (final Talk talk : talks.active()) {
+            final XML xml = talk.read();
+            final List<XML> requestNodes = xml.nodes("/talk/request");
+            final int current = idx + 1;
+            final String index = "index";
+            if (requestNodes.isEmpty()) {
+                talk.modify(new Directives().xpath("/talk").addIf("request")
+                        .attr(index, Integer.toString(current))
+                        .attr("id", this.generateRequestID()).add("type")
+                        .set(index).up().add("args")
+                );
+                idx = current;
+            }
+        }
+    }
+
+    /**
+     * Generates unique request ID value based on a {@link UUID} object.
+     *
```

Comment:
Empty line between method description and params is redundant (see methods in other classes).